CMakeLists.txt

```
@@ -24,11 +24,37 @@ if(NOT DEFINED PROJECT_VERSION)
     set(PROJECT_VERSION "6.0.0")
 endif()
 
-# Extract the major, minor, and patch versions
-string(REGEX MATCH "^[0-9]+\\.[0-9]+\\.[0-9]+" PROJECT_VERSION_BASE ${PROJECT_VERSION})
+# Remove the 'v' prefix if it exists and extract the major, minor, and patch versions
+string(REGEX MATCH "^[vV]?([0-9]+)\\.([0-9]+)\\.([0-9]+)(-([A-Za-z0-9]+))?" _ ${PROJECT_VERSION})
```

```suggestion
string(REGEX MATCH "^[vV]?([0-9]+)\\.([0-9]+)\\.([0-9]+)(-(\w+))?" _ ${PROJECT_VERSION})
```

Just using `\w` here is probably better. Someone will try to use other word characters then letters and numbers.

---

The next line also needs a check and has to fail if not the right amount of groups were matched.

---

just checked, this works now for versions with and without a pre-release thing